kernel/os/include/os/arch/sim/os/os_arch.h

```diff
@@ -60,6 +60,7 @@ typedef unsigned int os_stack_t;
 /* sim specific */
 struct stack_frame;
 void os_arch_frame_init(struct stack_frame *sf);
+void os_arch_frame_init_64bit(struct stack_frame *sf);
```

Comment:
probably would be better to have the same prototype regardless of arch and just use preprocessor to check that we build for supported arch (see https://github.com/apache/mynewt-core/pull/3042/commits/c384bec46f7d55339153b832f96b9d539fe7843b#r1265232905)